src/pages/chats/index.ts

```diff
@@ -1,65 +1,67 @@
-import { Block, TProps } from '../../utils/Block';
-import tpl from './tpl.hbs';
+import Block from '../../core/Block';
 import { Link } from '../../components/link';
-import { Dialogue } from '../../components/dialogues';
-import { MessagesBlock } from '../../components/messagesBlock';
-import { ChatForm } from '../../components/chatForm';
-
-import { dialogues } from '../../data';
-import { messages } from '../../data';
-
-import chatMenuDots from '../../../static/chat-menu-dots.svg';
+import { ChatsList } from '../../components/chatList';
+import { Messenger } from '../../components/messenger';
+import { ChatModal } from '../../components/chatModal';
+import { Button } from '../../components/button';
+import { Input } from '../../components/input';
 
+import tpl from './tpl.hbs';
 import './style.scss';
 
-export class Chat extends Block<TProps> {
-  constructor(props: TProps) {
-    super('div', props);
+export class ChatPage extends Block {
+  constructor() {
+    super({});
   }
 
   init() {
     this.children.link = new Link({
       value: 'Профиль >',
-      attr: {
-        href: '/profile',
-      },
+      to: '/settings',
     });
 
-    this.children.dialogues = dialogues.map(
-      (dialogue) =>
-        new Dialogue({
-          name: dialogue.user.name,
-          owner: dialogue.lastMessage.owner,
-          text: dialogue.lastMessage.text,
-          date: dialogue.lastMessage.date,
-          count: dialogue.newMessagesCount,
-          attr: {
-            class: 'dialogue',
-          },
-        })
-    );
+    this.children.chatsList = new ChatsList({});
 
-    this.children.messagesBlock = messages.map(
-      (messageBlock) =>
-        new MessagesBlock({
-          date: messageBlock.date,
-          messages: messageBlock.messages,
-          attr: {
-            class: 'messages',
-          },
-        })
-    );
+    this.children.messenger = new Messenger({});
 
-    this.children.chatForm = new ChatForm({
+    this.children.addChatsBtn = new Button({
+      value: 'Создать новый чат',
       attr: {
-        class: 'chat-right__form',
+        class: 'main-btn',
+      },
+      events: {
+        click: (e) => {
+          const target = e?.target as HTMLDivElement;
+          if (target && target.classList.contains('main-btn')) {
+            (this.children.chatModal as ChatModal)
+              .getContent()
+              ?.classList.add('modal--active');
+          }
+
+          (
+            (
+              (this.children.chatModal as ChatModal).children.input as Input
+            ).getContent() as HTMLInputElement
+          ).value = '';
+        },
```

Comment:
Тут я вижу нарушение инкапсуляции компонента ChatModal. Давай объясню: у ChatModal есть какая-то внутряняя структура, например поле Input. Только ChatModal должен знать про эту струкутуру и про то, как с ней взаимодействовать. Сейчас очищение input'a происходит вне ChatModal, получается, что сокрытие данных нарушено. Это решается следующим образом:

1. В классе ChatModal нужно сделать публичный метод, например clear:
```
  public clear() {
    ((this.children.input as Input).getContent() as HTMLInputElement).value = ''
  }
```

2. переписать этот кусок кода
```suggestion
              (this.children.chatModal as ChatModal).clear()
```

3. Повторить сделанное для кода ниже, который как я понял, активирует модалку

Reply:
Да тут фигню сделал, поправил в нескольких компонентах